lib/Interpreter/CIFactory.cpp

```diff
@@ -297,8 +318,18 @@ namespace {
 
     #if defined(__GLIBCXX__)
       // Avoid '__float128 is not supported on this target' errors
-      if (!opts.Language && !opts.StdVersion)
-        sArguments.addArgument("-std=c++11");
+      if (!opts.Language && !opts.StdVersion) {
+        switch (CxxStdCompiledWith()) {
+          case 17:
+            // Hopefully -D__float128=void can be removed when -std=c++17 works.
+            sArguments.addArgument("-D__float128=void");
```

Comment:
What is the problem this is solving? (I see an error message, but how to do get it?)

Do we need this even when compiling with clang? Do you have a bug tracker entry for GCC? As is I don't know what to make of this; e.g. https://godbolt.org/g/y9qoMM seems to work just fine with clang and C++1z on Linux. Does this problem only exist on OSX with libstdc++?

I'd really prefer *not* having this in here; `-D__float128=void` risks havoc that is greater than saying "it's a GCC bug" (if it is).